pybop/parameters/priors.py

```diff
@@ -199,3 +368,87 @@ def __init__(self, scale, loc=0, random_state=None):
         self.loc = loc
         self.scale = scale
         self.prior = stats.expon
+        self._n_parameters = 1
+
+    def _evaluateS1(self, x):
+        """
+        Evaluates the first derivative of the log exponential distribution at x.
+
+        Parameters
+        ----------
+        x : float
+            The point(s) at which to evaluate the first derivative.
+
+        Returns
+        -------
+        float
+            The value(s) of the first derivative at x.
+        """
+        log_pdf = self.__call__(x)
+        dlog_pdf = -1 / self.scale * np.ones_like(x)
+        return log_pdf, dlog_pdf
+
+
+class ComposedLogPrior(BasePrior):
```

Comment:
Let's rename and refactor `ComposedLogPrior` into `JointPrior` with `logpdf` and `logpdfS1` functions, analogous to `BasePrior`.

(In future, we could add functionality to marginalise the joint distribution to retrieve pdf's for the individual parameters - although the mapping would not be reversible if the parameters are correlated.)